client/webserver/api.go

```diff
@@ -189,6 +189,45 @@ func (s *WebServer) apiTrade(w http.ResponseWriter, r *http.Request) {
 	writeJSON(w, resp, s.indent)
 }
 
+// apiAccountExport is the handler for the '/account' API request.
+func (s *WebServer) apiAccountExport(w http.ResponseWriter, r *http.Request) {
+	form := new(accountExportAuthForm)
+	if !readPost(w, r, form) {
+		return
+	}
+	r.Close = true
+	account, err := s.core.AccountExport(form.Pass, form.Host)
+	if err != nil {
+		s.writeAPIError(w, "error exporting account: %v", err)
+		return
+	}
+	w.Header().Set("Connection", "close")
+	res := &struct {
+		OK      bool          `json:"ok"`
+		Account *core.Account `json:"account"`
+	}{
+		OK:      true,
+		Account: account,
+	}
+	writeJSON(w, res, s.indent)
```

Comment:
Also for followup, we should probably always indent this as it's a download that the user should be able to visually parse.

Reply:
It looks like the indent argument is ultimately ignored:

```
// writeJSON writes marshals the provided interface and writes the bytes to the
// ResponseWriter with the specified response code.
func writeJSONWithStatus(w http.ResponseWriter, thing interface{}, code int, indent bool) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	b, err := json.Marshal(thing)
	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
		log.Errorf("JSON encode error: %v", err)
		return
	}
	w.WriteHeader(code)
	_, err = w.Write(append(b, byte('\n')))
	if err != nil {
		log.Errorf("Write error: %v", err)
	}
}
```

I was able to indent successfully from a javascript change:

```
a.setAttribute('href', 'data:text/json,' + JSON.stringify(accountForExport, null, 2))
```